src/main/java/com/jme3/math/Quaternion.java

```diff
@@ -803,110 +803,114 @@ public float toAngleAxis(Vector3f axisStore) {
+
+		if (mag > 0.0f) {
+			float iMag = 1.0f / mag;
+			this.w = iMag * sw;
+			this.x = iMag * sx;
+			this.y = iMag * sy;
+			this.z = iMag * sz;
+
+		} else if (t < 0.5f) {
+			// this.w = q1.w;
+			// this.x = q1.x;
+			// this.y = q1.y;
+			// this.z = q1.z;
+		} else {
+			this.w = q2.w;
+			this.x = q2.x;
+			this.y = q2.y;
+			this.z = q2.z;
+		}
+
+		return this;
+	}
+
 	/**
-	 * <code>slerp</code> sets this quaternion's value as an interpolation
-	 * between two other quaternions.
-	 *
-	 * @param q1 the first quaternion.
-	 * @param q2 the second quaternion.
-	 * @param t the amount to interpolate between the two quaternions.
+	 * @deprecated Direct call to {@link #slerpLocal()}.
 	 */
-	public Quaternion slerp(Quaternion q1, Quaternion q2, float t) {
-		// Create a local quaternion to store the interpolated quaternion
-		if (q1.x == q2.x && q1.y == q2.y && q1.z == q2.z && q1.w == q2.w) {
-			this.set(q1);
-			return this;
-		}
+	public Quaternion slerp(Quaternion q2, float t) {
+		return this.slerpLocal(q2, t);
+	}
 
-		float result = (q1.x * q2.x) + (q1.y * q2.y) + (q1.z * q2.z) + (q1.w * q2.w);
+	/**
+	 * Sets the values of this quaternion to the slerp from itself to q2 by t
+	 *
+	 * @param q2 Final interpolation value
+	 * @param t The amount diffrence
+	 */
+	public Quaternion slerpLocal(Quaternion q2, float t) {
+		// make it nice and symmetrical
+		Quaternion q1 = this;
 
-		if (result < 0.0f) {
-			// Negate the second quaternion and the result of the dot product
-			q2.x = -q2.x;
-			q2.y = -q2.y;
-			q2.z = -q2.z;
-			q2.w = -q2.w;
-			result = -result;
-		}
+		float rw = q1.w * q2.w + q1.x * q2.x + q1.y * q2.y + q1.z * q2.z;
 
-		// Set the first and second scale for the interpolation
-		float scale0 = 1 - t;
-		float scale1 = t;
-
-		// Check if the angle between the 2 quaternions was big enough to
-		// warrant such calculations
-		if ((1 - result) > 0.0f) {
-			// Get the angle between the 2 quaternions,
-			// and then store the sin() of that angle
-			float theta = FastMath.acos(result);
-			float invSinTheta = 1f / FastMath.sin(theta);
-
-			// Calculate the scale for q1 and q2, according to the angle and
-			// its sine value
-			scale0 = FastMath.sin((1 - t) * theta) * invSinTheta;
-			scale1 = FastMath.sin((t * theta)) * invSinTheta;
+		if (rw < 0) {
+			return this.pureSlerpLocal(q2.normalizeLocal().negated(), t);
+		} else {
+			return this.pureSlerpLocal(q2, t);
 		}
+	}
 
-		// Calculate the x, y, z and w values for the quaternion by using a
-		// special
-		// form of linear interpolation for quaternions.
-		this.x = (scale0 * q1.x) + (scale1 * q2.x);
-		this.y = (scale0 * q1.y) + (scale1 * q2.y);
-		this.z = (scale0 * q1.z) + (scale1 * q2.z);
-		this.w = (scale0 * q1.w) + (scale1 * q2.w);
 
-		// Return the interpolated quaternion
-		return this;
+	public Quaternion pureSlerp(Quaternion q1, Quaternion q2, float t) {
+		return q1.clone().pureSlerpLocal(q2, t);
```

Comment:
```suggestion
		return set(q1).pureSlerpLocal(q2, t);
```

Methods like this should modify the quat they are called on.